lib/logstash/inputs/gelf.rb

```diff
@@ -96,7 +96,21 @@ def udp_listener(output_queue)
       # Gelfd parser outputs null if it received and cached a non-final chunk
       next if data.nil?
 
-      event = LogStash::Event.new(LogStash::Json.load(data))
+      begin
+        event = LogStash::Event.new(LogStash::Json.load(data))
+      rescue LogStash::Json::ParserError => e
+        @logger.info("JSON parse failure. Falling back to plain-text", :error => e, :data => data)
+        event = LogStash::Event.new("message" => data, "tags" => ["_jsonparsefailure"])
+      rescue StandardError => e
```

Comment:
Brilliant! Thanks for the reference.

My current problem is with the specs though. I've rebased my PR on current `master`. Had to move the exception handling to `LogStash::Inputs::Gelf.new_event`. Now my `output_queue` returns `nil`. Still trying to determine how to test this.